### PR TITLE
Add new WebKit Settings and enable MSE

### LINKED ISSUE #51: Problems with Twitch and YouTube
I can't watch any live streams on Twitch (Error Code 6000) or 1080p videos on YouTube. Missing "Media Source Extensions" and "MSE & WebM VP9".

### LINKED ISSUE #65: Smooth Scrolling
WebKit supports smooth scrolling on non-touch inputs (like mouse wheel and key presses). I should turn that on in Ephemeral.

### PR DESCRIPTION
**Pull request by cassidyjames**
Fixes #51; fixes #65

### CODE REVIEW

File: src/MainWindow.vala [hunk 1]
```diff
@@ -58,9 +58,28 @@ public class MainWindow : Gtk.Window {
         web_context.set_process_model (WebKit.ProcessModel.MULTIPLE_SECONDARY_PROCESSES);
         web_context.get_cookie_manager ().set_accept_policy (WebKit.CookieAcceptPolicy.NO_THIRD_PARTY);
 
+        var web_kit_settings = new WebKit.Settings();
+        web_kit_settings.allow_file_access_from_file_urls = true;
+        web_kit_settings.allow_universal_access_from_file_urls = false;
+        web_kit_settings.default_font_family = Gtk.Settings.get_default().gtk_font_name;
+        web_kit_settings.enable_dns_prefetching = true;
```
[the remark this answers is not in the document]
Comment (cassidyjames) [reply src/MainWindow.vala]: @arshubham thanks, that's a fair point! I was thinking about the potential loadtime speed up, but you're right in that it should not prefetch by default. If we add preferences at some point that could be a decent candidate.

Comment (cassidyjames) [inline src/MainWindow.vala]:
```suggestion
        web_kit_settings.enable_dns_prefetching = false;
```

**Comment by TingPing** [comment]
btw you'll *really* want 2.24 when it releases rather than ubuntu lts ships for MSE to function well.

File: src/MainWindow.vala [hunk 2]
```diff
@@ -58,9 +58,28 @@ public class MainWindow : Gtk.Window {
         web_context.set_process_model (WebKit.ProcessModel.MULTIPLE_SECONDARY_PROCESSES);
         web_context.get_cookie_manager ().set_accept_policy (WebKit.CookieAcceptPolicy.NO_THIRD_PARTY);
 
+        var web_kit_settings = new WebKit.Settings();
+        web_kit_settings.allow_file_access_from_file_urls = true;
+        web_kit_settings.allow_universal_access_from_file_urls = false;
+        web_kit_settings.default_font_family = Gtk.Settings.get_default().gtk_font_name;
+        web_kit_settings.enable_dns_prefetching = false;
```
Comment (TingPing) [inline src/MainWindow.vala]: A lot of these are just setting the default values, maybe not ideal.

Comment (cassidyjames) [reply src/MainWindow.vala]: @TingPing thanks, resolved in https://github.com/cassidyjames/ephemeral/commit/b6d04df55b5df97cddaf5c418deafd19f8de1399

**Pull request merged**